_posts/2017-02-05-Retrospective-Swift-at-Artsy.md

```diff
@@ -0,0 +1,337 @@
+
+In contrast, and this may be the last major time it happens, but people refer to the time it took to migrate [in][weeks1] [the][weeks2] [scale][weeks3] [of][weeks4] _weeks_ during the Swift 2 -> 3 migration. Having the language evolve is great, sometimes in ways that you [agree with][swift-api] and sometimes in ways [you don't][closed]. Being able to use your own version of your tools frees you to make it work for you and your business. We have been talking about [extending TypeScript][extend-ts] specifically for our applications.
+
+# React Native, one year later
+
+In our announcement we talked about the lack of nuanced post-mortems on React Native. We're now a year in, we can at least try to help out in that space. We're sticking with React Native for the foreseeable future. It would take some _drastic_ changes in the Apple ecosystem for us to re-consider this decision. So here's the summary after 1 year.
+
+* We can share concepts with web
+* Tools are built for our apps like ours
+* To do it right requires engineers willing to dive deep in JS
+* You need native experience to have a polished app
+* Dependency stack is still obscenely big
+* Opens native engineers to more projects, makes yours more welcoming to others
+* Problems do, and will occur, but everything is fixable by forking
+* Extensive communication with native code gets tricky to test and maintain
+* We ended up re-using quite a lot of existing native code
+* It makes working in native code feel like a bad chore for someone to do
+* Spending so much time in another environment will erode native knowledge
+* We're not going to re-write other existing Apps into React Native
+* Makes a lot of sense in an [additive approach][our-rn]
+
+So, should you use React Native? Maybe. *If you have an API driven app, probably.* 
+
+It's definitely worth a week of prototyping for any engineering team, then if that goes well you should look into GraphQL and Relay. They really makes React Native shine. 
+
+# Want to get started?
+
+- Here's the [official site][rn].
+- These [two][eggheads1] [series][eggheads2] are high-quality. I studied JavaScript by watching hours of egghead videos.
+- Run through the [f8 app][f8] series on  [makeitopen.com][f8-open].
+- Clone our React Native app, [Emission][emission].
+- Read the rest of our [series on React Native][series].
+- Read our summary on [our JavaScript stack choices][js-2017].
+
+If you'd like to look into GraphQL + Relay, but don't want to start building a server yourself, consider these GraphQL as a Services:
+
+- [scraphold](https://scaphold.io).
+- [graphcool](https://www.graph.cool/).
```

Comment:
Afaik @l2succes and @mzikherman were working on getting MP available to the public.